table/server/common/src/main/java/alluxio/table/common/udb/UdbFilterSpec.java

```diff
@@ -0,0 +1,505 @@
+            "Table %s is already specified with exact name, "
+                + "cannot have partition specification at the same time",
+            tableName);
+      }
+    }
+  }
+
+  public static UdbFilterSpec empty() {
+    return EMPTY_INSTANCE;
+  }
+
+  /**
+   * Checks if a table should be bypassed.
+   *
+   * @param tableName the table name
+   * @return true if the table is configured to be bypassed, false otherwise
+   * @see UdbFilterSpec#isFullyBypassedTable(String)
+   */
+  public boolean isBypassedTable(String tableName) {
+    boolean isSpecifiedByName = matchedByTemplates(mBypassedTables, tableName);
+    if (isSpecifiedByName) {
+      if (mTableBypassMode == Mode.EXCLUDE) {
+        return false;
+      }
+      return shadowedByIgnore(tableName, true);
+    }
+    boolean hasBypassedPartition = mBypassedPartitions.containsKey(tableName);
+    if (hasBypassedPartition) {
+      // no need to check mTableBypassMode here, bc if a table has partition specifications,
+      // it must be in inclusion mode (or none)
+      return shadowedByIgnore(tableName, true);
+    }
+    // the table is not listed in any way
+    if (mTableBypassMode == Mode.EXCLUDE) {
+      return shadowedByIgnore(tableName, true);
+    } else {
+      return false;
+    }
+  }
+
+  /**
+   * Checks if all partitions of a table should be bypassed.
+   *
+   * @param tableName the table name
+   * @return true if the table is configured to be fully bypassed, false otherwise
+   * @see UdbFilterSpec#isBypassedTable(String)
+   */
+  public boolean isFullyBypassedTable(String tableName) {
+    boolean isFullyBypassed = matchedByTemplates(mBypassedTables, tableName);
+    return shadowedByIgnore(tableName, handleInclusionExclusion(mTableBypassMode, isFullyBypassed));
+  }
+
+  /**
+   * Checks if a table is ignored.
+   *
+   * @param tableName the table name
+   * @return true if the table is ignored, false otherwise
+   */
+  public boolean isIgnoredTable(String tableName) {
+    boolean isIgnored = matchedByTemplates(mIgnoredTables, tableName);
+    return handleInclusionExclusion(mIgnoreMode, isIgnored);
+  }
+
+  /**
+   * Checks by a partition's name if it should be bypassed.
+   *
+   * @param tableName the table name
+   * @param partitionName the partition name
+   * @return true if the partition should be bypassed, false otherwise
+   */
+  public boolean isBypassedPartition(String tableName, String partitionName) {
+    if (isFullyBypassedTable(tableName)) {
+      return true;
+    }
+    if (!mBypassedPartitions.containsKey(tableName)) {
+      return false;
+    }
+    Mode partitionMode = mBypassedPartitions.get(tableName).getFirst();
+    boolean isBypassed =
+        matchedByTemplates(mBypassedPartitions.get(tableName).getSecond(), partitionName);
+    return shadowedByIgnore(tableName, handleInclusionExclusion(partitionMode, isBypassed));
+  }
+
+  private static boolean handleInclusionExclusion(Mode mode, boolean selected) {
```

Comment:
add some description here

Reply:
Fixed in cef77153696b4028a525829d6d17d898fa972e6c